File: spring-boot-swagger/src/main/java/com/example/swagger/api/FooApi.java
package com.example.swagger.api;

import io.swagger.annotations.*;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;

@RestController
@Api(value = "foo_v1")
public class FooApi {

    @ApiOperation(value = "findFoo", notes = "find Foo")
    @ApiResponses({
            @ApiResponse(code = 200, message = "OK"),
            @ApiResponse(code = 500, message = "Internal Server Error"),
            @ApiResponse(code = 404, message = "Not Found Foo")
    })
    @GetMapping("/foo/{id}")
    public String foo(@ApiParam(value = "foo ID", required = true) @PathVariable Long id) {
        return "foo";
    }
}

File: spring-boot-monitor/src/main/kotlin/com/example/demo/FooApi.kt
package com.example.demo

import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class FooApi {

    private val log: Logger = LoggerFactory.getLogger(this.javaClass)

    @GetMapping("/api/v1/foo")
    fun foo(): String {
        log.info("foo !!!!!!!!!")
        return "success"
    }
}